src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -222,6 +227,8 @@ private Symbol FindPropertyByName(TypeSymbol type, IdentifierNameSyntax name, re
                 }
             }
 
+            resultKind = lookupResult.Kind;
+            lookupResult.Free();
```

Comment:
It looks like we are leaking lookupResult when we are returning from inside of the loop above.